sled-agent/src/instance.rs

```diff
@@ -214,16 +217,22 @@ impl InstanceInner {
             action
         );
 
-        // Notify Nexus of the state change.
-        self.nexus_client
-            .cpapi_instances_put(
-                self.id(),
-                &nexus_client::types::InstanceRuntimeState::from(
-                    self.state.current(),
-                ),
-            )
-            .await
-            .map_err(|e| Error::Notification(e))?;
+        // If there's no migration, nothing further to consider in updating nexus.
+        // If we are migrating, the destination becomes the source of truth.
+        let update_nexus =
+            !was_migrating || dst_propolis_uuid == Some(*self.propolis_id());
```

Comment:
I left this comment in chat, but just to make sure it's public here too:

Rather than modifying the sled agent's behavior to self-filter, it seems like it would also be possible to propagate both the source and destination state changes back up to Nexus. If Nexus wanted to, it could make the decision about "when to perform the switch-over", and simply filter by the `propolis_id` to decide which is authoritative. This would still provide some context in the case of a failed / aborted migration to potentially roll back.

I don't think that necessarily needs to block this, but it's something we'll want to consider when we have real source/destination Propolis Instances and we want to carefully monitor both.

Reply:
Updated Nexus to handle this instead of Sled Agent.